src/components/Icon/Icon.tsx

```diff
@@ -219,42 +95,21 @@ function Icon({
     backdrop && styles.hasBackdrop,
   );
 
-  const defaultIconProps = {
-    className: styles.Svg,
-    focusable: 'false',
-    'aria-hidden': true,
-  };
-
-  if (untrusted) {
-    // eslint-disable-next-line no-console
-    console.warn(
-      'Deprecation: The untrusted prop is no longer needed, all strings passed into the Icon component are rendered as unsafe',
-    );
-  }
-
-  let SourceComponent = source as React.SFC<React.SVGProps<SVGSVGElement>>;
   let contentMarkup: React.ReactNode;
-  if (source === 'placeholder') {
-    contentMarkup = <div className={styles.Placeholder} />;
-  } else if (isBundledIcon(source)) {
-    SourceComponent = BUNDLED_ICONS[source];
-    const componentImportName = importForStringIcon(source);
-    // eslint-disable-next-line no-console
-    console.warn(
-      `Deprecation: passing the string "${source}" into <Icon source> is deprecated and will be removed in the next major version. Pass in the "${componentImportName}" React Component from the @shopify/polaris-icons package instead.`,
+  if (typeof source === 'function') {
+    const SourceComponent = source;
+    const sourceElement = (
+      <SourceComponent
+        className={styles.Svg}
+        focusable="false"
+        aria-hidden="true"
+      />
     );
-    contentMarkup = <SourceComponent {...defaultIconProps} />;
-  } else if (typeof source === 'function') {
-    const sourceElement = <SourceComponent {...defaultIconProps} />;
     if (React.isValidElement(sourceElement)) {
```

Comment:
Should we throw an error here?

Reply:
I don't think it's worth having this if check in the first place and that we should always do `contentMarkup = sourceElement` - if you pass a function that returns something that isn't a react element you deserve an explosion when trying to render it.

We don't need to do any error handling when "React can't render the result of that function" will happen anyway.

Reply:
I agree, but sometimes it's nice to throw an explicit error with a nicer error message than the default one. I'll leave it up to you

Reply:
Gonna kill it. In our most common use-case of internal use people will be using typescript which shall scowl at them before they even run the code.